server/settings.go

```diff
@@ -31,21 +40,48 @@ func (p *Plugin) settingsNotifications(header *model.CommandArgs, instanceID, ma
 	if connection.Settings == nil {
 		connection.Settings = &ConnectionSettings{}
 	}
-	connection.Settings.Notifications = value
+	switch args[1] {
+	case subCommandAssignee:
+		connection.Settings.SendNotificationsForAssignee = &value
+	case subCommandMention:
+		connection.Settings.SendNotificationsForMention = &value
+	case subCommandReporter:
+		connection.Settings.SendNotificationsForReporter = &value
+	case subCommandWatching:
+		connection.Settings.SendNotificationsForWatching = &value
+	default:
+		return p.responsef(header, helpText)
+	}
+
 	if err := p.userStore.StoreConnection(instanceID, mattermostUserID, connection); err != nil {
 		p.errorf("settingsNotifications, err: %v", err)
-		p.responsef(header, "Could not store new settings. Please contact your system administrator. error: %v", err)
+		p.responsef(header, errStoreNewSettings, err)
 	}
 
 	// send back the actual value
 	updatedConnection, err := p.userStore.LoadConnection(instanceID, mattermostUserID)
 	if err != nil {
-		return p.responsef(header, "Your username is not connected to Jira. Please type `jira connect`. %v", err)
+		return p.responsef(header, errConnectToJira, err)
 	}
 	notifications := settingOff
-	if updatedConnection.Settings.Notifications {
-		notifications = settingOn
+	switch args[1] {
+	case subCommandAssignee:
```

Comment:
Idem here, something like:

```go
settings := *updatedConnection.Settings

... 

if settings.NeedToSendNotificationsFor(role) { ... }
```

we would translate the subcommandRole to the role, but we can use a Golang map to do that (for example).

WDYT? Thank you! :-)

Reply:
It seems better and now I have implemented the same.